google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -145,16 +149,33 @@ public CancellableRunnable createDeployRunner(
         loggingHandler,
         deploymentConfiguration,
         targetEnvironment,
-        wrapCallbackForUsageTracking(callback, deploymentConfiguration, targetEnvironment));
+        wrapCallbackForUsageTracking(callback, targetEnvironment));
 
-    boolean isFlexCompat = targetEnvironment.isFlexible()
-        && AppEngineProjectService.getInstance().isFlexCompat(project, source);
-    if (targetEnvironment.isStandard() || isFlexCompat) {
-      return createStandardRunner(loggingHandler, Paths.get(source.getFilePath()), deploy,
-          isFlexCompat);
+    if (targetEnvironment.isStandard() || targetEnvironment.isFlexCompat()) {
```

Comment:
I think I know what you mean. Keen eye.

It was actually a design decision to start considering the environment as Standard/Flexible depending on the facet.

Compat has the peculiarity that it's Standard, but with a change in appengine-web.xml.

We definitely need to look at the appengine-web.xml here since, as explained below, we need to do a fresh check here. The problem is that if you save a deployment run configuration, which sets the environment, then manually change the appengine-web.xml file to make it a Compat (or Standard, works both ways) app, this won't be reflected in the already persisted configuration, so we need to check it here to avoid nasty errors.

The environment is also tied to the deployment source, not sure if I fully understand you there.